### PR TITLE
GS-8167 Migrate from jest to vitest

### PR DESCRIPTION
**Pull request by sandhya-spend**
[GS-8167](https://govspend.atlassian.net/browse/GS-8167)

Update repo to use `vitest` instead of `jest`.

Note:
`mobx` imports have the warning

```
stderr | packages/client/src/index.test.js
The MobX package does not have a default export. Use 'import { thing } from "mobx"' (recommended) or 'import * as mobx from "mobx"' instead."
```
This is resolved in a later version of `mobx` and needs to be separate task.

- [x] Check ci works well

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -1,35 +0,0 @@
-/* https://jestjs.io/docs/configuration */
-
-export default {
-  coverageProvider: 'v8',
-  coverageReporters: ['lcov'],
-  collectCoverageFrom: [
-    '<rootDir>/packages/*/src/**/*.js',
-    '!<rootDir>/packages/*/src/**/*.test.js',
-  ],
-  reporters: ['default', ['github-actions', { silent: false }]],
```
Comment (sandhya-spend) [inline jest.config.js:10]: In vite, `github-actions` is [default](https://vitest.dev/guide/reporters.html#github-actions-reporter).

File: jest.config.js [hunk 2]
```diff
@@ -1,35 +0,0 @@
-/* https://jestjs.io/docs/configuration */
-
-export default {
-  coverageProvider: 'v8',
-  coverageReporters: ['lcov'],
-  collectCoverageFrom: [
-    '<rootDir>/packages/*/src/**/*.js',
-    '!<rootDir>/packages/*/src/**/*.test.js',
```
Comment (sandhya-spend) [inline jest.config.js:8]: All test files are [excluded](https://vitest.dev/config/#include) from coverage.

**Comment by JBezerra** [comment]
Hey @sandhya-spend can we review that? If so, can you move to `Ready for review` ?

**Comment by sandhya-spend** [comment]
> Hey @sandhya-spend can we review that? If so, can you move to `Ready for review` ?

I am checking that the ci runs well, thats why its in draft.

File: jest.config.js [hunk 3]
```diff
@@ -1,35 +0,0 @@
-/* https://jestjs.io/docs/configuration */
-
-export default {
-  coverageProvider: 'v8',
```
Comment (sandhya-spend) [inline jest.config.js:4]: `v8` is the default [coverage provider](https://vitest.dev/config/#coverage-provider).

File: .github/workflows/pull-request.yaml [hunk 4]
```diff
@@ -35,9 +35,7 @@ jobs:
           prettier_command_prefix: yarn
 
       - name: Test
-        run: yarn run jest --ci --reporters github-actions --reporters summary --coverage .
-        env:
-          NODE_OPTIONS: '--no-warnings --experimental-vm-modules'
+        run: yarn run test  --reporter github-actions --reporter default  --coverage
```
Comment (stellarhoof) [inline .github/workflows/pull-request.yaml]: **suggestion:** Remove reporters configuration

`github-actions` is the default [when running in CI](https://vitest.dev/guide/reporters.html#github-actions-reporter) and `default` is well, the default.

File: packages/provider-mongo/test/globalSetup.js [hunk 5]
```diff
@@ -2,9 +2,17 @@
...
-export default async () => {
+export const setup = async () => {
   globalThis.__mongoServer = await MongoMemoryServer.create()
   // Each test file has its own environment with its own `global` instance so
   // we need to use the process environment.
-  process.env.MONGO_URI = global.__mongoServer.getUri()
+  process.env.VITE_MONGO_URI = global.__mongoServer.getUri()
+}
+
+// https://nodkz.github.io/mongodb-memory-server/docs/guides/integration-examples/test-runners/
```
Comment (stellarhoof) [inline packages/provider-mongo/test/globalSetup.js]: Here's the updated, working link: https://typegoose.github.io/mongodb-memory-server/docs/guides/integration-examples/test-runners

**Review by stellarhoof — APPROVED** [review]
- [x] Coverage is reported in PR
- [x] Tests run both globally and per package
 
🎉 🎉 🎉 Outstanding work! There was nothing I could find fault with this work, just a couple of comments.

The tests run even faster now and it on top of that, it allow us to more easily introduce TypeScript!

File: .github/workflows/pull-request.yaml (hunk 4 above)
Comment (sandhya-spend) [reply .github/workflows/pull-request.yaml]: Oops, missed it.

**Pull request merged**